app/routers/sessions.py

```diff
@@ -154,11 +154,34 @@ async def update_session(session_id: str, session_updates: UpdateSession):
         else:
             session_update_query["$set"].update({"events": [event_obj]})
     else:
-        session["events"].append(event_obj)
-        if "$push" not in session_update_query:
-            session_update_query["$push"] = {"events": event_obj}
+
+        if (
+            new_event == EventType.dummy_event
+            and session["events"][-1]["event_type"] == EventType.dummy_event
+        ):
+            # if previous event is dummy, just change the updated_at time of previous event
+            last_event_index = len(session["events"]) - 1
+            if "$set" not in session_update_query:
+                session_update_query["$set"] = {
+                    "events."
+                    + str(last_event_index)
+                    + ".updated_at": event_obj["created_at"]
```

Comment:
When we get the new event, we're calling it `new_event` and we do this `new_event = jsonable_encoder(session_updates)["event"]`.

Then we're creating a structured object out of it by doing
`event_obj = jsonable_encoder(Event.parse_obj({"event_type": new_event}))`

We should also name it as `new_event_obj` instead of `event_obj` just to keep things crystal clear

Reply:
done